File: ruby/easy/flip_game.rb
# @param {String} s
# @return {String[]}
def generate_possible_next_moves(s)
    res = Array.new
    for i in 0..s.length-2
        prefix = s.slice(0, i)
        suffix = s.slice(i+2, s.length)
        res.push((prefix ? prefix : '') + '--' + (suffix ? suffix : '')) if s[i] == '+' && s[i] == s[i+1]
    end
    return res
end

File: javascript/easy/flipGame.js
/**
 * @param {string} s
 * @return {string[]}
 */
var generatePossibleNextMoves = function(s) {
    const result = [];
    for(let i = 0; i < s.length-1; i++) {
        if(s[i] === '+' && s[i+1] === s[i]) {
            result.push(s.substring(0, i) + '--' + s.substring(i+2));
        }
    }
    return result;
};
